src/scipp/coords.py

```diff
@@ -2,247 +2,480 @@
-        self._memo = []  # names of product for cycle detection
-        self._aliases = []  # names that alias other names
-        self._consumed = []  # names that have been consumed
-        self._outputs = outputs  # names of outputs
-        self._graph = graph
-
-    def _add_coord(self, *, name):
-        if self._exists(name):
-            return _produce_coord(self.obj, name)
-        if isinstance(self._graph[name], str):
-            self._aliases.append(name)
-            out = {name: self._get_coord(self._graph[name])}
-            dim = (self._graph[name], )
-        else:
-            func = self._graph[name]
-            argnames = _argnames(func)
-            args = {arg: self._get_coord(arg) for arg in argnames}
-            have_all_dense_inputs = all([v[0] is not None for v in args.values()])
-            if have_all_dense_inputs:
-                dense_args = {k: v[0] for k, v in args.items()}
-                out = func(**dense_args)
-                if not isinstance(out, dict):
-                    out = {name: out}
-            else:
-                out = {}
-            have_event_inputs = any([v[1] is not None for v in args.values()])
-            if have_event_inputs:
-                event_args = {
-                    k: v[0] if v[1] is None else v[1]
-                    for k, v in args.items()
-                }
-                out_bins = func(**event_args)
-                if not isinstance(out_bins, dict):
-                    out_bins = {name: out_bins}
-                # Dense outputs may be produced as side effects of processing event
-                # coords.
-                for name in list(out_bins.keys()):
-                    if out_bins[name].bins is None:
-                        coord = out_bins.pop(name)
-                        if name in out:
-                            assert identical(out[name], coord)
-                        else:
-                            out[name] = coord
-            else:
-                out_bins = {}
-            out = {
-                k: (out.get(k, None), out_bins.get(k, None))
-                for k in list(out.keys()) + list(out_bins.keys())
-            }
-            dim = tuple(argnames)
-        self._rename.setdefault(dim, []).extend(out.keys())
-        for key, coord in out.items():
-            _store_coord(self.obj, key, coord)
-
-    def _exists(self, name):
-        in_events = self.obj.bins is not None and name in self.obj.bins.meta
-        return name in self.obj.meta or in_events
-
-    def _get_existing(self, name):
-        events = None if self.obj.bins is None else self.obj.bins.meta[name]
-        return self.obj.meta.get(name, None), events
-
-    def _get_coord(self, name):
-        if self._exists(name):
-            self._consumed.append(name)
-            if name in self._outputs:
-                return self._get_existing(name)
+    get_logger().info('Transforming coords %s -> %s%s%s%s', inputs_str, outputs_str,
```

Comment:
Should it be "Transforming coord" (no plural)?

Reply:
Why the old style `%s` notation by the way? Use f-strings? I didn't think logging html representations would be desirable here?

Reply:
> Should it be "Transforming coord" (no plural)?

I think plural is better because there are often several inputs and sometimes multiple outputs. I didn't want to implement proper singular/plural handling here...

> Why the old style %s notation by the way? Use f-strings? I didn't think logging html representations would be desirable here?

This is not about the HTML repr but laziness. Logging handlers format the messages if and only if they are not filtered out. So if there is no handler for level INFO, the message will never be formatted. And unfortunately, loggers use the old style syntax unless configured otherwise.
In this case, we probably don't save any real time doing this because formatting the message pieces is more work than gluing a few strings together...

Reply:
> loggers use the old style syntax unless configured otherwise.

Are you referring to Python loggers in general, or ours?

Reply:
In general. They are old as you can tell from the camelCase naming. And they use %-formatting by default. This can be changed to use {}-style. But that needs to be done with all handlers that are added to a logger which makes is harder for users to customise the scipp logger.

Reply:
👍 thanks, good to know.